checker/src/main/java/org/checkerframework/checker/resourceleak/ResourceLeakChecker.java

```diff
@@ -38,6 +48,70 @@ public ResourceLeakChecker() {}
    */
   public static final String COUNT_MUST_CALL = "countMustCall";
 
+  /**
+   * The exception types in this set are ignored in the CFG when determining if a resource leaks
+   * along an exceptional path. These kinds of errors fall into a few categories: runtime errors,
+   * errors that the JVM can issue on any statement, and errors that can be prevented by running
+   * some other CF checker.
+   */
+  private static final SetOfTypes DEFAULT_IGNORED_EXCEPTIONS =
+      SetOfTypes.anyOfTheseNames(
+          ImmutableSet.of(
+              // Any method call has a CFG edge for Throwable/RuntimeException/Error
+              // to represent run-time misbehavior. Ignore it.
+              Throwable.class.getCanonicalName(),
+              Error.class.getCanonicalName(),
+              RuntimeException.class.getCanonicalName(),
+              // Use the Nullness Checker to prove this won't happen.
+              NullPointerException.class.getCanonicalName(),
+              // These errors can't be predicted statically, so ignore them and assume
+              // they won't happen.
+              ClassCircularityError.class.getCanonicalName(),
+              ClassFormatError.class.getCanonicalName(),
+              NoClassDefFoundError.class.getCanonicalName(),
+              OutOfMemoryError.class.getCanonicalName(),
+              // It's not our problem if the Java type system is wrong.
+              ClassCastException.class.getCanonicalName(),
+              // It's not our problem if the code is going to divide by zero.
+              ArithmeticException.class.getCanonicalName(),
+              // Use the Index Checker to prevent these errors.
+              ArrayIndexOutOfBoundsException.class.getCanonicalName(),
+              NegativeArraySizeException.class.getCanonicalName(),
+              // Most of the time, this exception is infeasible, as the charset used
+              // is guaranteed to be present by the Java spec (e.g., "UTF-8").
+              // Eventually, this exclusion could be refined by looking at the charset
+              // being requested.
+              UnsupportedEncodingException.class.getCanonicalName()));
+
+  /**
+   * Command-line option for controlling which exceptions are ignored.
+   *
+   * @see #DEFAULT_IGNORED_EXCEPTIONS
+   * @see #getIgnoredExceptions()
+   */
+  public static final String IGNORED_EXCEPTIONS = "resourceLeakIgnoredExceptions";
+
+  /**
+   * A pattern that matches one or more consecutive commas, optionally preceded and followed by
+   * whitespace.
+   */
+  private static final Pattern COMMAS = Pattern.compile("\\s*(?:" + Pattern.quote(",") + "\\s*)+");
+
+  /**
+   * A pattern that matches an exception specifier for the {@link #IGNORED_EXCEPTIONS} option: an
+   * optional "? extends" followed by a qualified name. The whole thing can be padded with
```

Comment:
To me the `? extends` syntax is a bit verbose for a command-line option.  My intuition also says that most of the time, when someone specifies an exception type, they will likely mean that type and all its subtypes.  So maybe the default should be that writing a qualified name means all subtypes?  And for an exact type, maybe we could require `=fully.qualified.Name`?